src/lib/libfock/CDJK.cc

```diff
@@ -104,14 +116,14 @@ void CDJK::initialize_JK_core()
         }
     }
     timer_off("CD: schwarz");
-
     if (df_ints_io_ == "SAVE") {
-        // stick ncholesky in process environment for other codes that may use the integrals
-        Process::environment.globals["NAUX (SCF)"] = (double)ncholesky_;
-
         psio_->open(unit_,PSIO_OPEN_NEW);
+        psio_->write_entry(unit_, "length", (char*)&ncholesky_, sizeof(long int));
         psio_->write_entry(unit_, "(Q|mn) Integrals", (char*) Qmnp[0], sizeof(double) * ntri * ncholesky_);
         psio_->close(unit_,1);
+        // stick ncholesky in process environment for other codes that may use the integrals
+        // Not sure if this should really be here.  It is here because Ugur uses this option to get the number of cholesky vectors.  
+        Process::environment.globals["NAUX (SCF)"] = ncholesky_;
```

Comment:
I made the changes to Ugur's code.  I don't think I will make the changes to Eugene's code.  Ugur's was very simple because he was reading the same file whenever he wanted to open up cholesky.  I'm not really sure what is meant by NAUX(SCF) and NAUX(CC), so I will leave that code alone.